### ISSUE TITLE
create a way to run acceptance tests from folder trees directly in the source tree

### ISSUE THREAD

**Issue by RonnyPfannschmidt** [pytest-dev/pytest#2628]
our mechanism for acceptance testing is a disgrace, creating file trees in test scripts to run nested py.test instances against it is both wastefull, and hard to debug

instead we should have the file trees that show the issues inside of our source tree for direct use and examination

this also relates to #2207

**Comment by nicoddemus** [comment]
I agree having the explicit files would make things easier at first glance.

We would have to figure out how to specify some things which are needed by acceptance tests:

1. Which arguments to pass to pytest;
2. Should tests fail or pass, if so how many of each?
3. How to match against expected output.
4. Sometimes we run a "test suite" more than once, changing things in between runs and then asserting either status code, expected output, etc.
5. I'm sure I'm missing some other cases.

**Comment by RonnyPfannschmidt** [comment]
1. and 2. and 3. are part of handling this issue
4. is clearly a case where creating a copy of the file tree is sensible, i do wonder if we can track changes to be inflicted on those files in a better way
